CONTRIBUTING.md

```diff
@@ -23,21 +37,72 @@ The following is a check list of requirements that need to be satisfied in order
 * Notable changes (i.e. new features or changed behavior, bugfixes) are appropriately documented in CHANGELOG.md, functional changes also in godoc
```

Comment:
Aren't CHANGES.txt and CHANGELOG.md redundant?

Reply:
Hmm they are... @michaelsembwever do you have any suggestions regarding this? Is it ok if we just keep using `CHANGELOG.md` but new entries use JIRA issue IDs instead of GH pull request numbers? Otherwise we'd have to keep `CHANGELOG.md` around if anyone wants to look at the changelog for older releases and use `CHANGES.txt` for new releases.

Reply:
Oh in that case yeah we should probably keep using `CHANGELOG.md` but I'll see if I can get other opinions on this.

Reply:
updated `CONTRIBUTING.md` so references to `CHANGES.txt` are replaced with `CHANGELOG.md`